src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs

```diff
@@ -123,15 +126,16 @@ object Handler(TInput input, ILambdaContext context)
     /// </param>
     /// <returns>Task.</returns>
     public static Task TraceAsync<TInput>(
-        TracerProvider tracerProvider,
+        TracerProvider? tracerProvider,
         Func<TInput, ILambdaContext, Task> lambdaHandler,
         TInput input,
         ILambdaContext context,
         ActivityContext parentContext = default)
     {
         Guard.ThrowIfNull(lambdaHandler);
+        Guard.ThrowIfNull(context);
```

Comment:
It would probably be OK to check this only in TraceInternal to reduce the boilerplate.

Also please check if this now allows to remove some null-checks. I think `ILambdaContext?` should no longer appear anywhere since we never deal with null-contexts now.

Reply:
Also moved repeating context null-checks to internal methods.